### ISSUE TITLE
mysqli::real_connect may be called with null $flags

### ISSUE THREAD

**Issue by mondrake** [doctrine/dbal#4869]
### Bug Report

|    Q        |   A
|------------ | ------
| BC Break    | no
| Version     | 3.1

#### Summary

In PHP 8.1 you get a deprecation when `mysqli::real_connect` is called with null $flags:
`
mysqli::real_connect(): Passing null to parameter #7 ($flags) of type int is deprecated
`

**Comment by morozov** [comment]
Neither PHPStan nor Psalm in their current configuration catch this issue but PHPStan level 8 would:
```
$ phpstan a src/Driver/Mysqli/Connection.php
Note: Using configuration file phpstan.neon.dist.
 1/1 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100%

 ------ ----------------------------------------------------------------------------------- 
  Line   Connection.php                                                                     
 ------ ----------------------------------------------------------------------------------- 
  61     Parameter #7 $flags of method mysqli::real_connect() expects int, int|null given.  
 ------ ----------------------------------------------------------------------------------- 
                                                           
 [ERROR] Found 1 error

**Comment by greg0ire** [comment]
That would be since https://php.watch/versions/8.1/internal-func-non-nullable-null-deprecation (so indeed PHP 8.1, more precisely in https://github.com/php/php-src/commit/b10416a652d26577a22fe0b183b2258b20c8bb86).

I pushed a fix in https://github.com/doctrine/dbal/pull/4879, which merges up #4873 into 3.1.x

**Comment by iammati** [comment]
Is there maybe a patch/fix for `doctrine/dbal:^2.13.7` + PHP 8.1? Experiencing this at the very moment using a CMS (TYPO3 v11.5) which depends yet on dbal v2 (with deprecation exceptions reporting enabled)

**Comment by derrabus** [comment]
Please open a PR against the 2.13.x branch.
